### ISSUE TITLE
Function `remove` defined in the indexed-formula.js file works incorectly when passed an array.

### ISSUE THREAD

**Issue by Exulansis** [linkeddata/rdflib.js#145]
When the function receives an array as the argument, it calls itself once for each element of the array
```
if (st instanceof Array) {
  for (var i = 0; i < st.length; i++) {
    this.remove(st[i])
  }
  return this
}
```
The problem is that the function `this.remove` then calls the function ` this.removeStatement`, which then calls the function `RDFArrayRemove` (defined in the `util.js` file).
`rdfArrayRemove` uses the [array.splice](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) method which is destructive, and therefore it modifies the array passed to it from the `remove` function.

Now the second iterration of the loop in the `remove` function, the array we are iterating over has been modified(with element[i] removed). Trying to access element[i+1] will instead get us element[i+2], since the array has shrinked, and both the length and the position of elements has changed. Therefore the `remove` function ends up skipping elements.

Potential fix would be to pass a copy of the array rather than the reference:
```
if (st instanceof Array) {
      var copy = st.slice()
      for (var i = 0; i < copy.length; i++) {
        this.remove(copy[i])
      }
      return this
}
```

**Comment by timbl** [comment]
Good catch. This is only in the case that the array passed happens to be an index of the store.  Maybe the copy should be made only at that point
